lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -254,11 +310,109 @@ describe('BasicLtiLaunchApp', () => {
     });
   });
 
+  describe('concurrent fetching of groups and content', () => {
+    let contentUrlCall;
+    let groupsCall;
+
+    beforeEach(() => {
+      // Will attempt to fetch the 1. content url and 2. groups.
+      fakeConfig.api = {
+        authToken: 'dummyAuthToken',
+        viaCallbackUrl: 'https://lms.hypothes.is/api/files/1234',
+        sync: {
+          data: {
+            course: {
+              context_id: '12345',
+              custom_canvas_course_id: '101',
+            },
+          },
+          path: '/api/sync',
+        },
+      };
+      contentUrlCall = fakeApiCall.withArgs({
+        authToken: 'dummyAuthToken',
+        path: 'https://lms.hypothes.is/api/files/1234',
+      });
+
+      groupsCall = fakeApiCall.withArgs({
+        authToken: 'dummyAuthToken',
+        path: '/api/sync',
+        data: {
+          course: {
+            context_id: '12345',
+            custom_canvas_course_id: '101',
+          },
+        },
+      });
+    });
+
+    it('renders the spinner until both groups and contentUrl requests finish', async () => {
+      const contentUrl = contentUrlCall.resolves({
```

Comment:
I don't think `contentUrl` is what you are expecting it to be. `stub.resolves(...)` returns _the stub_ for chaining, not the Promise. In other words:

```js
const stub = sinon.stub()
stub.resolves('foobar') === stub // true. `resolves(...)` returns `this`.
```

Reply:
The same applies to other statements of the form:

```js
const someVar = someStub.resolves(...)
```

Below.

Reply:
To show what I mean, the tests still pass if I replace all uses of `await contentUrl` or `await groups` with `await true`:

```diff
diff --git a/lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js b/lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js
index be6044e6..948a64d4 100644
--- a/lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js
+++ b/lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js
@@ -347,16 +347,16 @@ describe('BasicLtiLaunchApp', () => {
     });
 
     it('renders the spinner until both groups and contentUrl requests finish', async () => {
-      const contentUrl = contentUrlCall.resolves({
+      contentUrlCall.resolves({
         via_url: 'https://via.hypothes.is/123',
       });
-      const groups = groupsCall.resolves(['group1', 'group2']);
+      groupsCall.resolves(['group1', 'group2']);
       const wrapper = renderLtiLaunchApp();
-      await contentUrl;
+      await true;
       // Spinner should not go away after first request
       wrapper.update();
       assert.isTrue(wrapper.find('Spinner').exists());
-      await groups;
+      await true;
       // Spinner should go away after the second request
       await waitForElementToRemove(wrapper, 'Spinner');
       // iframe should be visible
@@ -364,39 +364,39 @@ describe('BasicLtiLaunchApp', () => {
     });
 
     it('renders the iframe hidden after contentUrl succeeds and groups remains pending', async () => {
-      const contentUrl = contentUrlCall.resolves({
+      contentUrlCall.resolves({
         via_url: 'https://via.hypothes.is/123',
       });
       const wrapper = renderLtiLaunchApp();
-      await contentUrl;
+      await true;
       wrapper.update();
       assert.equal(wrapper.find('iframe').prop('style').visibility, 'hidden');
     });
 
     it('shows an error dialog if the first request fails and second succeeds', async () => {
-      const contentUrl = contentUrlCall.rejects(new ApiError(400, {}));
-      const groups = groupsCall.resolves(['group1', 'group2']);
+      contentUrlCall.rejects(new ApiError(400, {}));
+      groupsCall.resolves(['group1', 'group2']);
       const wrapper = renderLtiLaunchApp();
-      await contentUrl;
+      await true;
       // Should show an error after the first request fails
       await waitForElement(wrapper, 'FakeDialog[title="Authorize Hypothesis"]');
-      await groups;
+      await true;
       // Should still show an error even if the second request does not fail
       await waitForElement(wrapper, 'FakeDialog[title="Authorize Hypothesis"]');
     });
 
     it('shows an error dialog if the first request succeeds and second fails', async () => {
-      const contentUrl = contentUrlCall.resolves({
+      contentUrlCall.resolves({
         via_url: 'https://via.hypothes.is/123',
       });
-      const groups = groupsCall.rejects(new ApiError(400, {}));
+      groupsCall.rejects(new ApiError(400, {}));
       const wrapper = renderLtiLaunchApp();
-      await contentUrl;
+      await true;
       // Should not show an error yet
       assert.isFalse(
         wrapper.find('FakeDialog[title="Authorize Hypothesis"]').exists()
       );
-      await groups;
+      await true;
       // Should show an error after failure
       await waitForElement(wrapper, 'FakeDialog[title="Authorize Hypothesis"]');
     });
```